### PR TITLE
Add bash support

### PR DESCRIPTION
**Pull request by mehmetalici**
This PR adds bash support beside zsh. 

It does not modify file extensions as they are irrelevant. However, they should also be modified accordingly for convention.

### CODE REVIEW

**Comment by Luanf** [comment]
Thanks for the contribution! I understand there are some small changes required to work with bash.
However, there is an issue here: since it's common for ZSH users to have Bash installed too, but not so much the other way around, checking for ZSH_VERSION and BASH_VERSION is not the correct approach (a ZSH user can have a valid BASH_VERSION too but the bash conditional is always happening first).

Thus, I suggest we use ```current_shell=$(basename "$SHELL")``` instead, and apply the conditional around this variable:

```
current_shell=$(basename "$SHELL")

if [ "$current_shell" = "bash" ]; then
  echo "$BASH_VERSION"
  cat cligpt.zsh >> ~/.bashrc
elif [ "$current_shell" = "zsh" ]; then
  cat cligpt.zsh >> ~/.zshrc
else
  echo "Install unsuccessful. Shell is not supported."
  exit 1
fi
```

And same idea in the ```cligpt()``` function. 

Another thing is that seems you've changed the `read` function in the `zsh` path, which seems like a mistake, and should be reverted to the original. 

Thanks again!

**Review by Luanf — CHANGES_REQUESTED** [review]
Use current_shell=$(basename "$SHELL") instead of this conditionals.

File: cligpt.zsh [hunk 1]
```diff
@@ -38,7 +38,13 @@ cligpt() {
     echo "Output: $COMMAND"
   elif [ -n "$COMMAND" ]; then
     echo "-> $COMMAND"
-    read -q "REPLY?Do you want to run this command? (y/n)"
+    if [ -n "$BASH_VERSION" ]; then
+      read -r -p "Do you want to run this command? (y/n) "
+    elif [ -n "$ZSH_VERSION" ]; then
+      read -r -q "Do you want to run this command? (y/n) "
```
Comment (Luanf) [inline cligpt.zsh:44]: In the ZSH path, revert this read function back to the original.

Also use the same `current_shell` approach suggested in `install.sh`

**Comment by mehmetalici** [comment]
Hi Luan, thanks for the comments. 

Firstly, checking the variable `$BASH_VERSION` is the suggested method for checking which shell is currently running. 

Quoting [https://tldp.org/LDP/abs/html/internalvariables.html](https://tldp.org/LDP/abs/html/internalvariables.html):
```
Checking $BASH_VERSION is a good method of determining which shell is running. $SHELL does not necessarily give the correct answer.
```
Moreover, on a system where both bash and zsh is installed, only one of the variables will be defined in the environment of a given shell session.

Consider the following example:
```
      ▶  $ zsh -c 'echo Zsh version is $ZSH_VERSION; echo Bash version is $BASH_VERSION'
Zsh version is 5.8.1
Bash version is

      ▶  $ bash -c 'echo Zsh version is $ZSH_VERSION; echo Bash version is $BASH_VERSION'
Zsh version is
Bash version is 5.1.16(1)-release

      ▶  $ echo Zsh version is $ZSH_VERSION; echo Bash version is $BASH_VERSION
Zsh version is
Bash version is 5.1.16(1)-release
``` 

Secondly, `read` builtin command is not standardized and works differently in zsh and bash. For example:
``` 
      ▶  $ bash
      ▶  $ read -q 'Do you want to continue?'
bash: read: -q: invalid option
read: usage: read [-ers] [-a array] [-d delim] [-i text] [-n nchars] [-N nchars] [-p prompt] [-t timeout] [-u fd] [name ...]
``` 
This point is also discussed here: https://superuser.com/a/556006.

**Comment by mehmetalici** [comment]
There is currently a problem with the install.zsh. It has the shebang line `#/bin/bash` which sets the interpreter `bash` for the script. So the command `./install.zsh` issued in a zsh shell session will always call the bash interpreter for the file so the .zshrc will never gets modified. 

I think, it is a good practice to let the user decide how the function will be installed to their system. Therefore, install.sh can be simply removed and the instructions can be given in the README.md. 

For example, as a user, I might not want to append the whole function to my rc as it is quite lengthy and messes my rc. As a user, I would rather prefer moving the function to a conventional directory, such as  `~/.bash`, then in my `.bashrc` would do `source ~/.bash/cligpt.sh`, which is one-liner and therefore much cleaner. 

Alternatively, one can add --shell optional argument to install.sh accepting either bash or zsh strings. Install.sh would then install according to the given option. If no argument is given, it would install it for the default shell for the system (defined in the $SHELL environment variable).

**Comment by Luanf** [comment]
@mehmetalici 
I appreciate the time you've taken to provide this feedback!
Following your suggestions, I've come up with some improvements. 

You make a strong case for checking the $BASH_VERSION and $ZSH_VERSION variables instead of relying on $SHELL. I agree that this approach seems to be more reliable for determining the current shell session.

Regarding the read command, I've done some testing in both zsh and bash and found what worked for me was slightly different between the two.

I also see your point about the shebang line in install.zsh and how it's causing issues with the script. I don't want to remove it though - I still want to cater to users who are not comfortable doing more manual installations, if they are out there. So, I've attempted some code changes to it as well.

**I'm opening a Pull Request to your fork**, and if you are good with those changes we can merge and then I'll merge this PR. Feel free to suggest and comment on the PR, though. 

In case you don't want to bother, I'll probably merge it directly to master.

**Comment by mehmetalici** [comment]
I pulled in your PR to my fork (and resolved an outdated request change in this PR and I assume you need to approve it to unblock merging).

Thanks for your valuable cooperation :)

**Review by Luanf — APPROVED** [review]

**Pull request merged**